src/main/java/org/reaktivity/nukleus/tcp/internal/stream/ReadStream.java

```diff
@@ -85,6 +93,10 @@ int handleStream(
             // TCP reset
             handleIOExceptionFromRead();
         }
+        finally
+        {
```

Comment:
I think we should also key.clear(OP_READ); and register when group budget is available.